### PR TITLE
Support more common key names in native menus

### PR DESCRIPTION
**Pull request by RaymondLim**
This fixes issue https://github.com/adobe/brackets/pull/10249.

### CODE REVIEW

File: appshell/appshell_extensions_mac.mm [hunk 1]
```diff
@@ -1013,12 +1013,29 @@ NSUInteger processKeyString(ExtensionString& key)
...
+
+    unichar pageUpChar   = NSPageUpFunctionKey;
+    unichar pageDownChar = NSPageDownFunctionKey;
+    unichar homeChar     = NSHomeFunctionKey;
+    unichar endChar      = NSEndFunctionKey;
+    unichar insertChar   = NSHelpFunctionKey;
+    
     //replace special keys with ones expected by keyEquivalent
-    const ExtensionString del = (ExtensionString() += NSDeleteCharacter);
+    const ExtensionString pageUp    = (ExtensionString() += [[NSString stringWithCharacters: &pageUpChar length: 1] UTF8String]);
```
Comment (redmunds) [inline appshell/appshell_extensions_mac.mm]: @rlim Why can't the new keys be defined using same pattern as existing keys? For example:

```js
    const ExtensionString pageUp    = (ExtensionString() += NSPageUpFunctionKey);
```

Seems simpler and more consistent.

**Comment by redmunds** [comment]
@rlim Done with code review. Works great. Just 1 comment.

File: appshell/appshell_extensions_mac.mm (hunk 1 above)
Comment (RaymondLim) [inline appshell/appshell_extensions_mac.mm]: @redmunds Because they're `unichar` and we need to convert them to UTF8.

**Comment by redmunds** [comment]
Looks good. Merging.

**Pull request merged**